be/src/formats/parquet/group_reader.cpp

```diff
@@ -378,19 +378,35 @@ void GroupReader::_process_columns_and_conjunct_ctxs() {
     for (auto& column : _param.read_cols) {
         SlotId slot_id = column.slot_id();
         if (conjunct_ctxs_by_slot.find(slot_id) != conjunct_ctxs_by_slot.end()) {
+            bool can_use_dict_for_all_expressions = true;
             for (ExprContext* ctx : conjunct_ctxs_by_slot.at(slot_id)) {
                 std::vector<std::string> sub_field_path;
-                if (_try_to_use_dict_filter(column, ctx, sub_field_path, column.decode_needed)) {
-                    _use_as_dict_filter_column(read_col_idx, slot_id, sub_field_path);
-                } else {
-                    _left_conjunct_ctxs.emplace_back(ctx);
+                if (!_can_use_dict_filter(column, ctx, column.decode_needed)) {
```

Comment:
Can we just call `_try_to_use_dict_filter` first, and record correspond `subfiled`, once it returns `false`, we just skip this slotid, and if all calls return `true`, we call `_use_as_dict_filter_column` for every ctx and crrespond subfiled.

Reply:
if any of them returns false, we would also need to clear all `_dict_filter_ctx->conjunct_ctxs` which we already pushed before, while checking expressions which could use dict filter. Tried doing it before, looked much worse than current implementation

Reply:
@SHaaD94 maybe we don't need to clean it.

Reply:
I just think it's little cumbersome(can_use_dict_filter and then try_to_use_dict_filter), maybe we can do it in just one round. and could you add some test cases just like what we did in test/sql/test_hive_sink, 
and Left LGTM